### PR TITLE
HL-756 | Validate state_aid_max_percentage

### PR DESCRIPTION
**Pull request by rikuke**
## Description :sparkles:
The handler form fails silently if there is no option selected from the State Aid Maximum Percentage select element.
This PR extends the backend validations for the pay subsidy calculation to validate that the State Aid Maximum Percentage is selected and displays an error if is it not.

## Testing :alembic:
Navigate to the handler form, open an application for handling, try to create a calculation without selectin the State Aid Maximum Percentage.
## Screenshots :camera_flash:
Screenshot 2023-06-06 at 15 11 36

## Additional notes :spiral_notepad:
Investigating the issue, I noticed that there is some code on the frontend for validating the input fields, but enabling the frontend validation breaks the layout and the validations  as the date inputs use the same keys.
Fixing this is assigned to a separate ticket [HL-826](https://helsinkisolutionoffice.atlassian.net/browse/HL-826?atlOrigin=eyJpIjoiOTczYWI5ZTZlNWRhNGVjY2JlMWU3YzkwMGVjMThhYzkiLCJwIjoiaiJ9)

[HL-826]: https://helsinkisolutionoffice.atlassian.net/browse/HL-826?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: backend/benefit/calculator/api/v1/serializers.py [hunk 1]
```diff
@@ -272,6 +272,15 @@ def validate(self, data):
                 raise serializers.ValidationError(
                     {"end_date": _("End date cannot be empty")}
                 )
+            calculation = self.context["request"].data["calculation"]
```
Comment (mjturt) [inline backend/benefit/calculator/api/v1/serializers.py:275]: Seems that `request` is already in the context and `data` also maybe?

File: backend/benefit/calculator/api/v1/serializers.py [hunk 2]
```diff
@@ -272,6 +272,15 @@ def validate(self, data):
...
+            if calculation["state_aid_max_percentage"] is None:
```
Comment (mjturt) [inline backend/benefit/calculator/api/v1/serializers.py:276]: What about other falsy values like empty strings?

File: backend/benefit/calculator/api/v1/serializers.py (hunk 1 above)
Comment (rikuke) [reply backend/benefit/calculator/api/v1/serializers.py]: Screenshot 2023-06-13 at 11 48 25

Comment (rikuke) [reply backend/benefit/calculator/api/v1/serializers.py]: In this stage of executio, data does not have any knowledge of the calculation.

File: backend/benefit/calculator/api/v1/serializers.py [hunk 3]
```diff
@@ -272,6 +273,22 @@ def validate(self, data):
                 raise serializers.ValidationError(
                     {"end_date": _("End date cannot be empty")}
                 )
+            state_aid_max_input = self.context["request"].data["calculation"][
+                "state_aid_max_percentage"
+            ]
+            if (
```
Comment (mjturt) [inline backend/benefit/calculator/api/v1/serializers.py:279]: Could be written:

```
if (
    not state_aid_max_input or state_aid_max_input
    not in [choice[0] for choice in STATE_AID_MAX_PERCENTAGE_CHOICES]
):
    ...
```

or if the `STATE_AID_MAX_PERCENTAGE_CHOICES`  were defined as a Django enum, like something like this:
```
class STATE_AID_MAX_PERCENTAGE_CHOICES(models.IntegerChoices):
    FIFTY = 50, "50%"
    HUNDRED = 100, "100%"
```

then there would be attributes `.values` and `.choices` available and the statement could be written:
```
if not state_aid_max_input or state_aid_max_input not in STATE_AID_MAX_PERCENTAGE_CHOICES.values:
    ...
```
but… don't know if it's really worth the trouble.

File: backend/benefit/locale/fi/LC_MESSAGES/django.po [hunk 4]
```diff
@@ -512,6 +512,9 @@ msgstr "Aloituspäivä ei voi olla tyhjä"
 msgid "End date cannot be empty"
 msgstr "Päättymispäivä ei voi olla tyhjä"
 
+msgid "State aid maximum percentage cannot be empty"
```
Comment (mjturt) [inline backend/benefit/locale/fi/LC_MESSAGES/django.po:515]: Should there be translations for other languages too?

**Review by mjturt — APPROVED** [review]
Some minor nitpicks. Approving.

**Pull request merged**